modules/pwa/class.jetpack-pwa-manifest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+class Jetpack_PWA_Manifest {
+	/**
+	 * @var Jetpack_PWA_Manifest
+	 */
+	private static $__instance = null;
+
+	/**
+	 * When this query var is present, display the PWA manifest.
+	 *
+	 * @var string
+	 */
+	const PWA_MANIFEST_QUERY_VAR = 'jetpack_app_manifest';
+
+	/**
+	 * Singleton implementation
+	 *
+	 * @return Jetpack_PWA_Manifest
+	 */
+	public static function instance() {
+		if ( is_null( self::$__instance ) ) {
+			self::$__instance = new Jetpack_PWA_Manifest;
+		}
+
+		return self::$__instance;
+	}
+
+	/**
+	 * Registers actions the first time that instance() is called.
+	 */
+	private function __construct() {
+		add_action( 'wp_head', array( $this, 'render_manifest_link' ) );
+		add_action( 'amp_post_template_head', array( $this, 'render_manifest_link' ) );
+		add_action( 'template_redirect', array( $this, 'render_manifest_json' ), 2 );
+	}
+
+	function render_manifest_link() {
+		?>
+			<link rel="manifest" href="<?php echo esc_url_raw( $this->get_manifest_url() ); ?>">
+			<meta name="theme-color" content="<?php echo esc_attr( Jetpack_PWA_Helpers::get_theme_color() ); ?>">
+		<?php
+	}
+
+	public function get_manifest_url() {
+		return add_query_arg(
+			self::PWA_MANIFEST_QUERY_VAR, '1', home_url()
+		);
+	}
+
+	function render_manifest_json() {
+		// Do not load manifest in multiple locations
+		if ( is_front_page() && isset( $_GET[ self::PWA_MANIFEST_QUERY_VAR ] ) && $_GET[ self::PWA_MANIFEST_QUERY_VAR ] ) {
+			$theme_color = Jetpack_PWA_Helpers::get_theme_color();
+
+			$manifest = array(
+				'name'       => get_bloginfo( 'name' ),
+				'start_url'  => get_home_url(),
+				'short_name' => substr( get_bloginfo( 'name' ), 0, 12 ),
```

Comment:
I think it would be good to provide a way to filter the shortname.

Reply:
Further down, there is a filter called `jetpack_pwa_manifest` that a user could hook into and modify any key. Users should be able to change the short_name with something like this:

```php
add_filter( 'jetpack_pwa_manifest', function( $manifest ) {
    $manifest['short_name'] = 'Hello World';
    return $manifest;
} );
```

Reply:
I agree it's probably better to have one filter for the whole manifest, to save running a separate filter for every value.